src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -80,7 +80,11 @@ public class HttpUtil {
 
   public static CloseableHttpClient getHttpClient() {
     if (httpClient == null) {
-      initHttpClient();
+      synchronized (HttpUtil.class) {
+        if (httpClient == null) {
+          initHttpClient();
+        }
+      }
```

Comment:
do we need outer check of `(httpClient == null)`

Reply:
Outer check is not required for correctness, but it reduces lock contention. Lock is only required during initialisation; once it is initialised, no lock is needed anymore. But if the outer `if` was not there, lock would have to be acquired every time even when the initialisation is long finished.

Reply:
True that, thanks Lukas!